### DISCUSSION TITLE
issue with country flag after removing country code

### DISCUSSION THREAD

**Discussion by dmatysiak94** [typesnippet/antd-phone-input#77] [General]
I aslo would like to report bug that input shows country ISO code and phone code instead of the flag if the user clears the Phone value after search using. 

https://github.com/typesnippet/antd-phone-input/assets/37019712/02a9581a-315d-4420-ad33-f6ec7f5f5af4

**Comment by Set27** [comment]
@dmatysiak94 do you find any solution? I have similar:

https://github.com/typesnippet/antd-phone-input/assets/49077110/aef27433-c57b-4716-ac0c-5c90e5991370

here is code to reproduce for maintainers:
```js
<Question title="Phone" name="phone">
        <PhoneInput country={"us"} onlyCountries={["us", "ca"]} />
</Question>
```

**Comment by ArtyomVancyan** [comment]
@dmatysiak94, @Set27, the issue you're showcasing was occurring in older versions. Please make sure you're using the latest versions of [antd-phone-input](https://www.npmjs.com/package/antd-phone-input) and [react-phone-hooks](https://www.npmjs.com/package/react-phone-hooks).

**Reply by Set27** [reply]
@ArtyomVancyan I use 
```
    "antd": "^5.16.0",
    "antd-phone-input": "^0.3.5",
```

**Reply by Set27** [reply]
I also get error in console
```
⨯ node_modules/react-phone-hooks/styles.js (11:0) @ injectStyles
 ⨯ ReferenceError: document is not defined
```

**Reply by ArtyomVancyan** [reply]
Please install the latest version of `react-phone-hooks` manually and try again. I think the problem is with the package manager when determining what dependency version should be installed.

**Reply by ArtyomVancyan** [reply]
Also, are you using server-side rendering (SSR)?

**Reply by Set27** [reply]
@ArtyomVancyan install `react-phone-hooks`
```
    "@ant-design/nextjs-registry": "^1.0.0",
    "antd": "^5.16.0",
    "antd-phone-input": "^0.3.5",
    "next": "14.1.4",
    "react": "^18",
    "react-dom": "^18",
    "react-phone-hooks": "^0.1.2",
```

> Also, are you using server-side rendering (SSR)?

I am not really sure, how to check it (just start with Next js).

**Reply by ArtyomVancyan** [reply]
Thanks for mentioning **Next.js**. Yes, it is a framework for static website generation from React-based applications with server-side rendering. That's why you get the `ReferenceError: document is not defined`. It's natural for Next.js. I am not sure how to deal with importing dynamic styles in SSR.

**Reply by ArtyomVancyan** [reply]
P.S. The usage is correct and should not cause issues with an ordinary React application.

**Reply by Set27** [reply]
this article is help to avoid error in console, but does not solve problem with number. Anyway want to left if here
[https://nextjs.org/docs/app/building-your-application/rendering](https://nextjs.org/docs/app/building-your-application/rendering)

**Reply by ArtyomVancyan** [reply]
Have you tried using the `"use client"` convention to fix your problem?

**Reply by Set27** [reply]
@ArtyomVancyan I've tried. It does not help

**Reply by ArtyomVancyan** [reply]
@Set27, @dmatysiak94 I have released the new version that should fix the problem. Can you please check?

**Reply by Set27** [reply]
@dmatysiak94 the same `Internal error: ReferenceError: document is not defined`

**Reply by ArtyomVancyan** [reply]
Thanks for your time @dmatysiak94. Can you please share the repo link or create one with minimal usage where the error is reproduced so I can dive in and fix the issue?

**Reply by ArtyomVancyan** [reply]
@dmatysiak94, please ensure your application uses `0.3.8` version of the [package](https://www.npmjs.com/package/antd-phone-input).

**Reply by SyedZainullahQazi** [reply]
I'm facing the same issue, Plus is there a way that we can make the input unique for a single country that doesn't allows any other country input? if not then I would love to contribute to the repo.

**Reply by ArtyomVancyan** [reply]
Hi @SyedZainullahQazi, we could not find the case when it occurred. Can you provide the exact steps to reproduce the issue?

> Is there a way that we can make the input unique for a single country that doesn't allow any other country input?

If you mean only one country must be allowed, then yes, you can use the `onlyCountries` property and specify the ISO code of the country.

Everyone is welcome to contribute to any of my open-source projects if there is an idea, suggestion, improvement, bug fix... you are welcome.

**Comment by ArtyomVancyan** [comment]
@dmatysiak94, @Set27, and @SyedZainullahQazi, thanks for pointing out the issue. Finally, I could find out the cause and fix it. You can upgrade the version of the package to use the latest version where it is fixed.
